Build fail on CentOS 5.11: Absoft (ASM symbols missing)

MTT link from last night showing the problem: https://mtt.open-mpi.org/index.php?do_redir=2279

```
../../../opal/.libs/libopen-pal.so: undefined reference to `__sync_sub_and_fetch_4'
./../../opal/.libs/libopen-pal.so: undefined reference to `__sync_bool_compare_and_swap_4'
libopen-pal.so: undefined reference to `__sync_add_and_fetch_4'
: undefined reference to `__sync_add_and_fetch_8'
```

@hjelmn Can you have a look?

@hjelmn This problem is still occurring on master.  Can you please have a look?  https://mtt.open-mpi.org/index.php?do_redir=2288

Absoft's build system is a 32-bit platform, GCC: 4.1.2, Absoft: Absoft 32-bit Pro Fortran 16, CentOS5.11.

```
$ ./configure FC=af90 FCFLAGS=-Z790 "CFLAGS=-g -pipe" --enable-picky --enable-debug --enable-mpirun-prefix-by-default
```

The error is:

```
../../../opal/.libs/libopen-pal.so: undefined reference to `__sync_sub_and_fetch_4'
./opal/.libs/libopen-pal.so: undefined reference to `__sync_bool_compare_and_swap_4'
reference to `__sync_add_and_fetch_4'
./../opal/.libs/libopen-pal.so: undefined reference to `__sync_add_and_fetch_8'
```

--- a/config/opal_config_asm.m4
+++ b/config/opal_config_asm.m4
@@ -88,7 +88,9 @@ AC_DEFUN([OPAL_CHECK_SYNC_BUILTIN_CSWAP_INT128], [
 AC_DEFUN([OPAL_CHECK_SYNC_BUILTINS], [
   AC_MSG_CHECKING([for __sync builtin atomics])
 
-  AC_TRY_COMPILE([], [__sync_synchronize()],
+  AC_TRY_LINK([long tmp;], [__sync_synchronize();
+__sync_bool_compare_and_swap(&tmp, 0, 1);
+__sync_add_and_fetch(&tmp, 1);],
     [AC_MSG_RESULT([yes])
      $1],
     [AC_MSG_RESULT([no])

